Library/ViewModels/LiveStreamContainerPageViewModelTests.swift

```diff
@@ -0,0 +1,174 @@
+    let chatPage = LiveStreamContainerPage.chat(project: .template, liveStreamEvent: .template)
+
+    self.vm.inputs.willTransition(toPage: chatPage)
+    self.vm.inputs.pageTransition(completed: true)
+
+    self.chatButtonTextColor.assertValues([.ksr_grey_500, .white])
+    self.infoButtonTextColor.assertValues([.white, .ksr_grey_500])
+
+    self.vm.inputs.willTransition(toPage: infoPage)
+    self.vm.inputs.pageTransition(completed: true)
+
+    self.chatButtonTextColor.assertValues([.ksr_grey_500, .white, .ksr_grey_500])
+    self.infoButtonTextColor.assertValues([.white, .ksr_grey_500, .white])
+  }
+
+  func testFonts() {
+    self.chatButtonTitleFontName.assertValueCount(0)
+    self.infoButtonTitleFontSize.assertValueCount(0)
+    self.infoButtonTitleFontName.assertValueCount(0)
+    self.infoButtonTitleFontSize.assertValueCount(0)
+
+    self.vm.inputs.configureWith(project: .template, liveStreamEvent: .template,
+                                 refTag: .projectPage, presentedFromProject: true)
+    self.vm.inputs.viewDidLoad()
+
+    self.vm.inputs.didLoadViewControllersIntoPagesDataSource()
+
+    let infoPage = LiveStreamContainerPage.info(project: .template, liveStreamEvent: .template,
+                                                refTag: .projectPage, presentedFromProject: true)
+    let chatPage = LiveStreamContainerPage.chat(project: .template, liveStreamEvent: .template)
+
+    self.chatButtonTitleFontName.assertValues([".SFUIText"])
+    self.chatButtonTitleFontSize.assertValues([14])
+    self.infoButtonTitleFontName.assertValues([".SFUIText-Semibold"])
+    self.infoButtonTitleFontSize.assertValues([14])
+
+    self.vm.inputs.willTransition(toPage: chatPage)
+    self.vm.inputs.pageTransition(completed: true)
+
+    self.chatButtonTitleFontName.assertValues([".SFUIText", ".SFUIText-Semibold"])
+    self.chatButtonTitleFontSize.assertValues([14, 14])
+    self.infoButtonTitleFontName.assertValues([".SFUIText-Semibold", ".SFUIText"])
+    self.infoButtonTitleFontSize.assertValues([14, 14])
+
+    self.vm.inputs.willTransition(toPage: infoPage)
+    self.vm.inputs.pageTransition(completed: true)
+
+    self.chatButtonTitleFontName.assertValues([".SFUIText", ".SFUIText-Semibold", ".SFUIText"])
+    self.chatButtonTitleFontSize.assertValues([14, 14, 14])
+    self.infoButtonTitleFontName.assertValues([".SFUIText-Semibold", ".SFUIText", ".SFUIText-Semibold"])
+    self.infoButtonTitleFontSize.assertValues([14, 14, 14])
+  }
+
+  func testPagedToPage() {
+    self.pagedToPage.assertValueCount(0)
+    self.pagedToPageDirection.assertValueCount(0)
+
+    self.vm.inputs.configureWith(project: .template, liveStreamEvent: .template,
+                                 refTag: .projectPage, presentedFromProject: true)
+    self.vm.inputs.viewDidLoad()
+
+    self.vm.inputs.didLoadViewControllersIntoPagesDataSource()
+
+    XCTAssertTrue(self.pagedToPage.values[0].isInfoPage)
```

Comment:
we shouldn't reach into the test observers `values` like this. in fact it should be private.

instead we should construct the values we want to assert against.